test/instrumentation/context.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+var test = require('tape')
+var { getHTTPDestination } = require('../../lib/instrumentation/context')
+
+test('#getHTTPDestination', function (t) {
+  t.test('username and pass', (t) => {
+    const url = 'http://user:pass@testing.local:1234/path?query'
+    t.deepEqual(getHTTPDestination(url, 'external'), {
+      service: {
+        name: 'http://testing.local:1234',
+        resource: 'testing.local:1234',
+        type: 'external'
+      },
+      address: 'testing.local',
+      port: 1234
+    })
+    t.end()
+  })
+
+  t.test('https default port', (t) => {
+    const url = 'https://www.elastic.co:443/products/apm'
```

Comment:
Shouldn't this url not have a port if we were to test the default port fallback?

```suggestion
    const url = 'https://www.elastic.co/products/apm'
```

Reply:
Yeah, I will add a test with and without default port to show the diff.